### PR TITLE
Fix/esxi

### PR DESCRIPTION
**Pull request by rainzm**
**这个 PR 实现什么功能/修复什么问题**:
1. Remove 'Destory' operation that is unnecessary for VirtualDiskManager
2. Separate detach disk and delete disk when rebuilding.

    In previous versions, set
    `removeSpec.FileOperation = types.VirtualDeviceConfigSpecFileOperationDestroy'
    to delete disk indirectly.

    But, when its parent has only one child, the parent will be deleted along with it.
    And the consequence is failure to reinstall the system.

    Now, detach disk without deleteing backing file and then remove backing
    file via datastore's operation.
3. Modify the disk size correctly when creating vm
4. Detach all disk first when deleting vm to prevent to delete delta disk's parent disk.

针对问题2：

之前为了修复振华的问题做的修改：#4994
现在来看没有测试以下情况：image_cache中的某个image (简称IM） 只有一个vm在用，重装vm就会触发问题，因为 IM 会被删除掉。

**是否需要 backport 到之前的 release 分支**:
- release/3.0
- release/3.1
- release/3.2

/area esxiagent
/cc @swordqiu @zexi

### CODE REVIEW

File: pkg/hostman/diskutils/vddk.go [hunk 1]
```diff
@@ -155,7 +155,12 @@ func (vd *VDDKDisk) MountRootfs() fsdriver.IRootFsDriver {
 		log.Errorf("VDDKDisk Mount failed: %s", err)
 	}
 	// something is wrong
-	vd.UmountRootfs(nil)
+	if vd.Proc != nil {
+		err := vd.Proc.Kill()
+		if err != nil {
+			log.Errorf("")
```
Comment (wanyaoqi) [inline pkg/hostman/diskutils/vddk.go:161]: log.Errorf(err)

File: pkg/multicloud/esxi/host.go [hunk 2]
```diff
@@ -1035,7 +1030,16 @@ func (host *SHost) CloneVM(ctx context.Context, from *SVirtualMachine, ds *SData
...
 	}
 
-	return NewVirtualMachine(host.manager, &moVM, host.datacenter), nil
+	// resize the disk
+	vm := NewVirtualMachine(host.manager, &moVM, host.datacenter)
+	sort.Sort(byDiskType(vm.vdisks))
+	for i, s := range newSizes {
+		err := vm.vdisks[i].Resize(ctx, s)
+		if err != nil {
+			log.Errorf("no.%d vdisk.Resize failed", i)
```
Comment (ioito) [inline pkg/multicloud/esxi/host.go:1039]: 可以把 err 也打出来

File: pkg/hostman/diskutils/vddk.go (hunk 1 above)
Comment (rainzm) [reply pkg/hostman/diskutils/vddk.go]: fix

File: pkg/multicloud/esxi/host.go (hunk 2 above)
Comment (rainzm) [reply pkg/multicloud/esxi/host.go]: fix

**Pull request merged**